src/strategies/snet-liquidity-providers/index.ts

```diff
@@ -0,0 +1,76 @@
+import { BigNumber, BigNumberish } from '@ethersproject/bignumber';
+import { formatUnits } from '@ethersproject/units';
+import { Multicaller, call } from '../../utils';
+
+export const author = 'bonustrack';
+export const version = '0.1.1';
+
+type FinalResult = [
+  Record<string, BigNumberish>,
+  Record<string, BigNumberish>,
+  BigNumberish
+];
+
+const erc20Abi = [
+  'function balanceOf(address account) external view returns (uint256)',
+  'function totalSupply() external view returns (uint)'
+];
+
+const parseNumber = (value: BigNumberish): BigNumber => BigNumber.from(value);
+
+const computeTokenContribution = (
+  lpBalance: BigNumber,
+  lpTotalSupply: BigNumber,
+  contractTokenBalance: BigNumber
+) => {
+  lpTotalSupply = lpTotalSupply.isZero() ? BigNumber.from(1) : lpTotalSupply;
+  const tokenContribution = lpBalance
+    .mul(contractTokenBalance)
+    .div(lpTotalSupply);
+  return tokenContribution;
+};
+
+const multiCallerFactory = (network, provider, blockTag) => (abi) =>
+  new Multicaller(network, provider, abi, { blockTag });
+
+export async function strategy(
+  space,
+  network,
+  provider,
+  addresses,
+  options,
+  snapshot
+): Promise<Record<string, number>> {
+  const blockTag = 'latest';
```

Comment:
@Vivek205 block can't be latest always - it may affect the voting power in future and change the results of closed proposals, It should be something like this:
```
  const blockTag = typeof snapshot === 'number' ? snapshot : 'latest';
```

Reply:
Hi @ChaituVR , this is expected behaviour, we would like to have it updated for every new block.

Reply:
@Vivek205 hmm but this is not the way strategies works, we need a way to get consistent results once the proposal is closed else if voting power keeps changing in the future it could break things ( also it could be hard for you to make decisions based on something that is changing in future - ex: if I vote with 1 VP and in future, I may buy 100 VP and change the results)

For your requirement, the good solution (for now) would be to use snapshot input in the proposal creation page
https://snapshot.org/#/snet.eth/create?snapshot=1
![image](https://user-images.githubusercontent.com/15967809/144004987-12778f71-0070-44fe-acb4-ae44b1c63931.png)

You can give a snapshot block that is expected to be completed on proposal closure would give you correct voting power

Reply:
fixed in https://github.com/snapshot-labs/snapshot-strategies/pull/202/commits/0025f5fe0aad6ead0eae45ca1fc52914a094bc35